client/src/data_layer_configs/lead_and_copper_violations_by_county_config.ts

```diff
@@ -1,48 +1,61 @@
-import { DataLayer, DataSourceType, GeoJsonDataLayer, LegendInfo } from '@/model/data_layer';
-import { FillLayer } from 'mapbox-gl';
+import { DataSourceType, GeoJsonDataLayer, LegendInfo, PopupInfo } from '@/model/data_layer';
+import { Expression, FillLayer } from 'mapbox-gl';
+import { colorMapToBuckets } from '@/util/data_layer_util';
 
 const ID: string = 'epa-lead-and-copper-violations';
 
-const LEGEND_COLOR_MAP = {
-  0: '#FFEAE5',
-  25: '#FFAB99',
-  50: '#FF9680',
-  75: '#FF8166',
-  100: '#FF5733',
+const LEGEND_COLOR_MAPPING = [
+  0,
+  '#FFEAE5',
+  25,
+  '#FFAB99',
+  50,
+  '#FF9680',
+  75,
+  '#FF8166',
+  100,
+  '#FF5733',
+];
+
+/**
+ * Mapbox expression which interpolates pairs of bucket 'stops' + colors to produce continuous
+ * results for the map.
+ *
+ *  See https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/#interpolate.
+ */
+const leadAndCopperViolationsInterpolation: Expression = [
+  'interpolate',
+  ['linear'],
+  ['get', 'Lead and Copper Rule'],
+  ...LEGEND_COLOR_MAPPING,
+];
+
+const legendInfo: LegendInfo = {
+  title: 'Number of violations',
+  bucketMap: colorMapToBuckets(LEGEND_COLOR_MAPPING),
 };
 
-const STYLE_LAYER: FillLayer = {
+const styleLayer: FillLayer = {
   id: `${ID}-style`,
   source: ID,
   type: 'fill',
   paint: {
-    'fill-color': [
-      'interpolate',
-      ['linear'],
-      ['get', 'Lead and Copper Rule'],
-      0,
-      LEGEND_COLOR_MAP[0],
-      25,
-      LEGEND_COLOR_MAP[25],
-      50,
-      LEGEND_COLOR_MAP[50],
-      75,
-      LEGEND_COLOR_MAP[75],
-      100,
-      LEGEND_COLOR_MAP[100],
-    ],
+    'fill-color': leadAndCopperViolationsInterpolation,
     'fill-opacity': 0.75,
     'fill-outline-color': '#B2391F',
   },
   layout: {
-    // Make the layer visible by default.
-    visibility: 'visible',
+    // Make the layer hidden by default.
+    visibility: 'none',
   },
 };
 
-const LEGEND_INFO: LegendInfo = {
-  title: 'Number of violations',
-  bucketMap: new Map(Object.entries(LEGEND_COLOR_MAP)),
+const popupInfo: PopupInfo = {
+  title: 'County',
+  subtitle: '320 estimated lead service lines',
```

Comment:
is `320` significant here? or a placeholder?

Reply:
Placeholder. Left a TODO to replace.